### ISSUE TITLE
IndexOutOfBoundsException

### ISSUE THREAD

**Issue by alexdumitru89** [onyxbits/raccoon4#62]
Hello,

When I try to download an app from CLI I keep getting this error:

```
java.lang.IndexOutOfBoundsException: Index: 0
        at java.util.Collections$EmptyList.get(Collections.java:4454)
        at com.akdeniz.googleplaycrawler.GooglePlay$AndroidAppDeliveryData.getDownloadAuthCookie(GooglePlay.java:5981)
        at com.akdeniz.googleplaycrawler.DownloadData.<init>(DownloadData.java:35)
        at com.akdeniz.googleplaycrawler.GooglePlayAPI.delivery(GooglePlayAPI.java:464)
        at com.akdeniz.googleplaycrawler.GooglePlayAPI.purchaseAndDeliver(GooglePlayAPI.java:476)
        at de.onyxbits.raccoon.cli.Play.downloadApp(Play.java:223)
        at de.onyxbits.raccoon.cli.Router.main(Router.java:182)
        at de.onyxbits.raccoon.Main.main(Main.java:59)
!fail.Index: 0!
```

**Comment by onyxbits** [comment]
Am 03.06.2019 14:42, schrieb Alex:
>
> Hello,
>
> When I try to download an app from CLI I keep getting this error:
>
> |java.lang.IndexOutOfBoundsException: Index: 0
>          at java.util.Collections$EmptyList.get(Collections.java:4454)
>          at com.akdeniz.googleplaycrawler.GooglePlay$AndroidAppDeliveryData.getDownloadAuthCookie(GooglePlay.java:5981)
>          at com.akdeniz.googleplaycrawler.DownloadData.<init>(DownloadData.java:35)
>          at com.akdeniz.googleplaycrawler.GooglePlayAPI.delivery(GooglePlayAPI.java:464)
>          at com.akdeniz.googleplaycrawler.GooglePlayAPI.purchaseAndDeliver(GooglePlayAPI.java:476)
>          at de.onyxbits.raccoon.cli.Play.downloadApp(Play.java:223)
>          at de.onyxbits.raccoon.cli.Router.main(Router.java:182)
>          at de.onyxbits.raccoon.Main.main(Main.java:59)
> !fail.Index: 0!
> |

|How about some more details? e.g. which Raccoon version, which app 
(version) and commandline parameters?|

**Comment by alexdumitru89** [comment]
I'm using version 4.7 and it does it on any app.

**Comment by onyxbits** [comment]
Am 03.06.2019 18:52, schrieb Alex:
>
> I'm using version 4.7 and it does it on any app.
>
>

Can't confirm this. Check if it works with the GUI. Chances are that the 
account got logged out.

**Comment by thecoderok** [comment]
I am seeing same issue as well. This happens only for some apps. I see error for fighting.stickmanandgun.zombieshooter.

Command (GUI):
```
$ java -jar /Users/vmganzha/Downloads/raccoon-4.14.0.jar
```
Command line:
```
$ java -jar /Users/vmganzha/Downloads/raccoon-4.14.0.jar  --gpa-download fighting.stickmanandgun.zombieshooter
java.lang.IndexOutOfBoundsException: Index: 0
	at java.util.Collections$EmptyList.get(Collections.java:4454)
	at com.akdeniz.googleplaycrawler.GooglePlay$AndroidAppDeliveryData.getDownloadAuthCookie(GooglePlay.java:5981)
	at com.akdeniz.googleplaycrawler.DownloadData.<init>(DownloadData.java:35)
	at com.akdeniz.googleplaycrawler.GooglePlayAPI.delivery(GooglePlayAPI.java:464)
	at com.akdeniz.googleplaycrawler.GooglePlayAPI.purchaseAndDeliver(GooglePlayAPI.java:476)
	at de.onyxbits.raccoon.cli.Play.downloadApp(Play.java:223)
	at de.onyxbits.raccoon.cli.Router.main(Router.java:182)
	at de.onyxbits.raccoon.Main.main(Main.java:59)
!fail.Index: 0!
```

But it works fine for other apps, e.g.:
```
$ java -jar /Users/vmganzha/Downloads/raccoon-4.14.0.jar  --gpa-download com.aurecas.stickmanzombieshooter
/Users/vmganzha/Raccoon/content/apps/com.aurecas.stickmanzombieshooter/com.aurecas.stickmanzombieshooter-300295.apk
#######....
/Users/vmganzha/Raccoon/content/apps/com.aurecas.stickmanzombieshooter/config.armeabi_v7a-300295.apk
###########....
```

**Comment by thecoderok** [comment]
DownloadAuthCookie seem to be empty when it tries to read first item:

![image](https://user-images.githubusercontent.com/3173477/86411505-d939ed80-bc71-11ea-98a2-68c9cfdf6fcd.png)

**Comment by onyxbits** [comment]
Yep, I'll have a closer look at this tomorrow.

**Comment by onyxbits** [comment]
Should be fixed in v4.15
